src/clj_kondo/impl/core.clj

```diff
@@ -408,21 +411,26 @@
                     (schedule ctx {:filename fn
                                    :uri (->uri nil nil fn)
                                    :source (slurp file)
-                                   :lang (lang-from-file path default-language)}
+                                   :lang (lang-from-file path default-language)
+                                   :total-files (+ total-files 1)}
                               dev?)))))
             ;; assume directory
-            (run! #(schedule ctx (assoc % :lang (lang-from-file (:filename %) default-language)) dev?)
-                  (sources-from-dir ctx file canonical?)))
+            (let [sources (sources-from-dir ctx file canonical?)]
+              (run! #(schedule ctx (assoc %
+                                          :lang (lang-from-file (:filename %) default-language)
+                                          :total-files (+ total-files (count sources))) dev?)
+                    sources)))
           (= "-" path)
           (schedule ctx {:filename (or filename "<stdin>")
                          :source (slurp *in*)
                          :lang (if filename
                                  (lang-from-file filename default-language)
                                  default-language)} dev?)
           (classpath? path)
-          (run! #(process-file ctx % default-language canonical? filename)
-                (str/split path
-                           (re-pattern path-separator)))
+          (let [paths (str/split path
+                                 (re-pattern path-separator))]
+            (run! #(process-file ctx % default-language canonical? filename (count paths))
```

Comment:
It seems this count is supposed to be total-files, but it's actually the number of classpath entries here, so jars + directories.

Reply:
yeah, indeed, any suggestion how to get all possible files here?

Reply:
Also, I'm ok on we don't have all specific files, but something roughly, a different name would be better though if that was the case

Reply:
In parallel mode you don't know the number of files before hand. So I don't think we can reliably add this number at all.

Reply:
RIght, I think I can try to make that on lsp side, probably will be easier and enough for what we want to fix: a better feedback, it doesn't need to be totally precise. I let you know if that works and then change this PR